instrumentation/spring/spring-batch-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/event/CustomSpanEventTest.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.batch.v3_0.event;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.javaagent.instrumentation.spring.batch.v3_0.runner.JobRunner;
+import io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions;
+import io.opentelemetry.sdk.testing.assertj.TraceAssert;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+abstract class CustomSpanEventTest {
+  private final JobRunner runner;
+
+  @RegisterExtension
+  static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  public CustomSpanEventTest(JobRunner runner) {
+    this.runner = runner;
+  }
+
+  @Test
+  void should_be_able_to_call_Span_current___and_add_custom_info_to_spans() {
+    runner.runJob("customSpanEventsItemsJob");
+
+    testing.waitAndAssertTraces(
+        trace ->
+            itemSpans(
+                trace.hasSpansSatisfyingExactly(
+                    span ->
+                        span.hasName("BatchJob customSpanEventsItemsJob")
+                            .hasKind(SpanKind.INTERNAL)
+                            .hasEventsSatisfyingExactly(
+                                event -> event.hasName("job.before"),
+                                event -> event.hasName("job.after")),
+                    span ->
+                        span.hasName("BatchJob customSpanEventsItemsJob.customSpanEventsItemStep")
+                            .hasKind(SpanKind.INTERNAL)
+                            .hasParent(trace.getSpan(0))
+                            .satisfies(
+                                spanData -> {
+                                  // CompositeChunkListener has broken ordering that causes
+                                  // listeners that do not override order() to appear first at all
+                                  // times
+                                  // because of that a custom ChunkListener will always see a Step
+                                  // span when using spring-batch versions [3, 4)
+                                  // that bug was fixed in 4.0
+                                  if (VERSION_GREATER_THAN_4_0) {
+                                    OpenTelemetryAssertions.assertThat(spanData)
```

Comment:
could use static imports for these
```suggestion
                                    assertThat(spanData)
```